packages/mdast-util-split/src/index.js

```diff
@@ -58,22 +29,21 @@ class Splitter {
   }
 
   visit (node, index, parent) {
-    // when we are at root
     if (!parent) {
+      // we are at the root
       return
     }
-    if ((node.type !== 'heading' || node.depth !== this.depth) &&
-      parent.type === 'root' && this.lastIndex === -1) {
-      this.introduction.children.push(node)
-      return
-    }
+
     if (node.type === 'heading' && node.depth === this.depth) {
```

Comment:
l.70 `(node.type === 'heading' && node.depth === this.depth)` was the inverse of l.65 `(node.type !== 'heading' || node.depth !== this.depth)`